sycl/plugins/unified_runtime/ur/adapters/cuda/common.cpp

```diff
@@ -0,0 +1,111 @@
+//===--------- common.cpp - CUDA Adapter -----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include "common.hpp"
+
+#include <cuda.h>
+
+#include <sstream>
+
+ur_result_t map_error_ur(CUresult result) {
```

Comment:
We're trying to keep the scope of this PR and the porting to effort to just a straight port of the existing code to avoid any changes to the behavior of the plugin/adapter, so I don't think it makes sense to do it at this stage. Plus the size of the PR means that reviewing any actual functional changes at the same time would be tricky.

It also requires a resolution to https://github.com/oneapi-src/unified-runtime/issues/500